seqr/views/apis/report_api.py

```diff
@@ -344,9 +344,13 @@ def _process_saved_variants(saved_variants_by_family, family_individual_affected
                 potential_com_het_gene_variants[gene_id].append(variant)
         for gene_id, comp_het_variants in potential_com_het_gene_variants.items():
             if len(comp_het_variants) > 1:
+                if len(comp_het_variants) > 2:
+                    variant['discovery_notes'] = 'The following variants are part of the multinucleotide variant {}: {}'.format(
+                        variant['variantId'], ', '.join([v['variantId'] for v in comp_het_variants ]))
+                else:
+                    variant['inheritance_models'] = {'AR-comphet'}
```

Comment:
this used to be inside the `for variant in comp_het_variants:` loop. This change needs to be inside that loop as well. Since the discovery note is the same for all the variants, I think I would still create the note here, and then set it below in the loop
```
discovery_notes = None
if len(comp_het_variants) > 2:
    discovery_notes = 'The following ...'.format(...)
...
for variant in comp_het_variants:
    if discovery_notes:
        variant['discovery_notes'] = discovery_notes
    else:
        variant['inheritance_models'] = {'AR-comphet'}
```

Reply:
In the code above, I think the variants in `comp_het_variants` belong to a gene. Do they also belong to an MNV? I'm trying to find an MNV in the `comp_het_variants` as the parent variant and the rest of variants as the nested variants of the MNV. But when I ran the code with project `R0384_rare_genomes_project_gen`, I got a `comp_het_variants` with these variants: [19-1912633-G-T, 19-1912632-GC-TT, 19-1912632-GC-TT]. It seems missing `19-1912634-C-T` and duplicating `19-1912632-GC-TT`. Does it have the right variants?

Reply:
Or, should I check the tags for the variants in the `comp_het_variants` to find MNVs?

Reply:
Ah I see the problem - the issue is my instructions to you were wrong. Sadly, we can't just repurpose `comp_het_variants` here. What we need to do is where we define `potential_com_het_gene_variants` we also need to define another dict for `potential_mnvs`, which will be defined as variants in the same genes with the same tag, and then explicitly check those. So it will be something like
```
potential_com_het_gene_variants = defaultdict(list)
potential_mnvs = defaultdict(lambda: defaultdict(list))
for variant in saved_variants:
    variant['main_transcript'] = _get_variant_main_transcript(variant)
    ...
    for gene_id in variant_json['transcripts'].keys():
        for tag in variant_json['discovery_tag_names']:
            potential_mnvs[gene_id][tag].append(variant_json)
mnv_genes = set()
for gene_id, variants_by_tag in potential_mnvs.items():
    mnvs = next((variants for variants in variants_by_tag.values() if len(variants) > 2), None)
    if not mnvs:
        continue
    mnv_genes.add(gene_id)
    discovery_note = <construct note from mnvs>
    for variant in mnvs:
        variant['discovery_note'] = discovery_note
for gene_id, comp_het_variants in potential_com_het_gene_variants.items():
    if gene_id in mnv_genes:
        continue
    if len(comp_het_variants) > 1:
    ...
```

Reply:
let me know if you would rather meet to discuss

Reply:
Thanks for your detailed instructions. I prefer to do it by myself first and then we meet later so that I can understand better the logic among the data. Maybe we can meet sometime next week.